core/src/main/java/org/infinispan/commands/LocalFlagAffectedCommand.java

```diff
@@ -17,14 +19,22 @@
     * not be modified directly, only via the {@link #setFlags(Set)}, {@link #addFlag(Flag)} and {@link
     * #addFlags(Set)} methods.
     */
-   Set<Flag> getFlags();
+   default Set<Flag> getFlags() {
+      return EnumUtil.enumSetOf(getFlagsBitSet(), Flag.class);
+   }
+
+   long getFlagsBitSet();
```

Comment:
If this is supposed to be backwards-compatible, then you need to have a default implementation for `getFlagsBitSet()` based on `getFlags()` now, and only switch them in 9.0.